File: Frontend/src/events/converters/card-battle.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { CardBattleDTO } from "@/events/dto/card-battle.dto";
import type { CardBattle } from "@/models";

export class CardBattleConverter {
  public static convert(cardBattleDto: DeepRequired<CardBattleDTO>): CardBattle {
    return {
      id: cardBattleDto.id,
    };
  }
}

File: Backend/Events/Converters/CardBattleConverter.cs
using Backend.Domain.Models;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class CardBattleConverter
{
    public static CardBattleDTO ToDTO(CardBattle cardBattle) => new()
    {
        OpponentId = cardBattle.OpponentId,
    };
}
